mlflow/transformers/peft.py

```diff
@@ -0,0 +1,50 @@
+"""
+PEFT (Parameter-Efficient Fine-Tuning) is a library for efficiently adapting large pretrained
+models without fine-tuning all of model parameters but only a small number of (extra) parameters.
+Users can define a PEFT model that wraps a Transformer model to apply a thin adapter layer on
+top of the base model. The PEFT model provides almost the same APIs as the original model such
+as from_pretrained(), save_pretrained().
+"""
+_PEFT_ADAPTOR_DIR_NAME = "peft"
+
+
+def is_peft_model(model) -> bool:
+    try:
+        from peft import PeftModel
+    except ImportError:
+        return False
+
+    return isinstance(model, PeftModel)
+
+
+def get_peft_base_model(model):
+    """Extract the base model from a PEFT model."""
+    peft_config = model.peft_config.get(model.active_adapter) if model.peft_config else None
```

Comment:
Is `model.peft_config` a dict? Does it always exist? If so, can we remove `if model.peft_config else None`?

Reply:
ah I see, it can be None?

Reply:
It's a bit confusing but `model.peft_config` is a dictionary that maps adapter name like "lora" to `PeftConfig` object. As far as I read [existing adapter classes](https://github.com/huggingface/peft/tree/main/src/peft/tuners), the property should not be None, but just wanted to be safe as there is no validation for it to be non-null.